src/GraphQL.Tests/NameFieldResolverTests.cs

```diff
@@ -14,27 +16,38 @@ public class NameFieldResolverTests
         [InlineData("fullInfo", "Anyone 20")]
         [InlineData(null, null)]
         [InlineData("unknown", "", true)]
-        [InlineData("FullInfoWithParam", "", true)]
-        public void resolve_should_work_with_properties_and_methods(string name, object expected, bool throws = false)
+        [InlineData("FullInfoWithParam", "test Anyone 20")]
+        [InlineData("FULLINFOWITHPARAM", "test Anyone 20")]
+        [InlineData("FullInfoWithContext", "Anyone 20")]
+        [InlineData("FromService", "hello")]
+        [InlineData("AmbiguousExample", "", true)]
+        public async Task resolve_should_work_with_properties_and_methods(string name, object expected, bool throws = false)
         {
             var person = new Person
             {
                 Age = 20,
                 Name = "Anyone"
             };
 
-            Func<object> result = () => NameFieldResolver.Instance.ResolveAsync(
+            var services = new ServiceCollection();
+            services.AddSingleton(new Class1());
+            Func<ValueTask<object>> result = () => NameFieldResolver.Instance.ResolveAsync(
                 new ResolveFieldContext
                 {
                     Source = person,
                     FieldDefinition = new GraphQL.Types.FieldType { Name = name },
-                    FieldAst = new GraphQLField { Name = name == null ? default : new GraphQLName(name) }
-                }).GetAwaiter().GetResult();
+                    FieldAst = new GraphQLField { Name = name == null ? default : new GraphQLName(name) },
+                    Arguments = new Dictionary<string, ArgumentValue>()
+                    {
+                        { "prefix", new ArgumentValue("test ", ArgumentSource.Literal) }
+                    },
+                    RequestServices = services.BuildServiceProvider(),
+                });
 
             if (throws)
-                Should.Throw<InvalidOperationException>(() => result());
+                await Should.ThrowAsync<InvalidOperationException>(async () => await result());
```

Comment:
Test message text too.

Reply:
This is a `[Theory]` and the messages vary depending on the failure.  In general, I really have no inclination to verify exception messages, and I do not believe that verifying the message adds value to the testing.  Changing an exception message is not generally considered a breaking change.  Besides, some exception messages change with the framework in use or language.  You may rework this method and add validation of exception messages if you choose to do so.

Reply:
Exception messages are useful to see what situation happened (for test reader). If messages vary I use `Contains`.